Add audio comparison subcommand

AdamVenn opened the pull request:
Hi there! I added a command to compare two audio files, using a little of the code (with permission) in https://github.com/benthevining/plugin-cancellation-testing.

I'd be keen to continue to develop some more functionality around audio comparison/analysis for this project. Let me know if the PR needs some modifications before merging.

AdamVenn committed:
Update to JUCE 8.0.10

AdamVenn committed:
Update to CLI11 2.5.0

AdamVenn committed:
Update to json 3.12.0

AdamVenn committed:
Update gitignore

AdamVenn committed:
Update C++ version

AdamVenn committed:
Add AudioDiff subcommand

AdamVenn committed:
Add JSON output format to list parameters

AdamVenn committed:
Update to JUCE 8.0.12

AdamVenn committed:
Github action

AdamVenn committed:
Build text format from json

AdamVenn committed:
Add extra parameter info

CrushedPixel commented:
Hello Adam, thanks for your contribution to this project. I've reviewed the PR and it looks good - apart from the GH Actions which I have no experience with and therefore can't say much about. But as long as they work for you I'll be happy to include them.

Truth be told, I'm not actively working with JUCE (or C++, for that matter) anymore, therefore I'm not particularly motivated to actively maintain this project right now. Since your PR looks good and you are motivated to add more related features, would you like to become a maintainer and continue at your own discretion?

If so, I'll make you a maintainer and you can merge. If not, I'll merge and continue reviewing what you bring along, although I can't really guarantee timely responses.

AdamVenn commented:
Yeah, I'd be very happy to be a maintainer! Thank you! I appreciate your reviewing.

AdamVenn committed:
Use new createWriterFor

JUCE deprecated a function which used raw pointers so we now use the newer, unique_ptr-based function.

AdamVenn committed:
Compiler warning: make internal function static

AdamVenn committed:
Compiler warning: initialize variable

AdamVenn committed:
Compiler warning: floating point comparison

AdamVenn committed:
Compiler warning: integer types
